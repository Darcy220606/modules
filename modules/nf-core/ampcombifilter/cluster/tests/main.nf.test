nextflow_process {

    name "Test Process AMPCOMBIFILTER_CLUSTER"
    script "../main.nf"
    process "AMPCOMBIFILTER_CLUSTER"

    tag "modules"
    tag "modules_nfcore"
    tag "ampcombifilter"
    tag "ampcombifilter/cluster"
    tag "ampcombifilter/complete"

    setup {
        run("AMPCOMBIFILTER_COMPLETE") {
            script "../../../ampcombifilter/complete/main.nf"
            process {
                """
                input[0] =
                    [
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombifilter/sample_1_ampcombi.tsv', checkIfExists: true),
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombifilter/sample_2_ampcombi.tsv', checkIfExists: true)
                    ]
                """
            }
        }
    }

    test("ampcombifilter_cluster - metagenome") {
        when {
            process {
                """
                input[0] = AMPCOMBIFILTER_COMPLETE.out.tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.cluster_tsv[0]).readLines()[0].contains("Linear/Cyclic/Branched"),
                    file(process.out.rep_cluster_tsv[0]).readLines()[0].contains("total_cluster_members"),
                    process.out.versions).match() }
            )
        }
    }

    test("ampcombifilter_cluster - metagenome - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = AMPCOMBIFILTER_COMPLETE.out.tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
