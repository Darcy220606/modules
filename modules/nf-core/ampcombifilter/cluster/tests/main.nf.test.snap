{
    "ampcombifilter_cluster - metagenome - stub": {
        "content": [
            {
                "0": [
                    "Ampcombi_summary_cluster.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "Ampcombi_summary_cluster_representative_seq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "Ampcombi_cluster.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "3": [
                    "versions.yml:md5,c2a55e88850ad92558a10e81aed6ed35"
                ],
                "cluster_tsv": [
                    "Ampcombi_summary_cluster.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "log": [
                    "Ampcombi_cluster.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "rep_cluster_tsv": [
                    "Ampcombi_summary_cluster_representative_seq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,c2a55e88850ad92558a10e81aed6ed35"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T09:50:23.018800346"
    },
    "ampcombifilter_cluster - metagenome": {
        "content": [
            true,
            true,
            [
                "versions.yml:md5,c2a55e88850ad92558a10e81aed6ed35"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T09:49:54.685331442"
    }
}